nymea-gpio-tool/application.cpp

```diff
@@ -0,0 +1,46 @@
+#include "application.h"
+
+#include <QDebug>
+#include <signal.h>
+
+static void catchUnixSignals(const std::vector<int>& quitSignals, const std::vector<int>& ignoreSignals = std::vector<int>())
+{
+    auto handler = [](int sig) ->void {
+        switch (sig) {
+        case SIGQUIT:
+            qDebug() << "Cought SIGQUIT quit signal...";
+            break;
+        case SIGINT:
+            qDebug() << "Cought SIGINT quit signal...";
+            break;
+        case SIGTERM:
+            qDebug() << "Cought SIGTERM quit signal...";
+            break;
+        case SIGHUP:
+            qDebug() << "Cought SIGHUP quit signal...";
+            break;
+        case SIGSEGV: {
+            qCritical() << "Cought SIGSEGV signal. Segmentation fault!";
+            exit(EXIT_FAILURE);
+        }
+        default:
+            break;
+        }
+
+        Application::quit();
+    };
+
+    // all these signals will be ignored.
+    for (int sig : ignoreSignals)
+        signal(sig, SIG_IGN);
+
+    for (int sig : quitSignals)
+        signal(sig, handler);
```

Comment:
Please add some comment on why this signal stuff is necessary... It is not obvious to me what this is needed for and actually looks wrong to intentionally break UNIX process communication standards... After all, there is a reason why those signals exist and behave that way.

Reply:
this is a command line tool, which normally gets closed using `ctrl +c`. Without this interrupt handler, it would not clean up the code properly, and the Gpio's would remain in the system. With catching the unix signals, i can cleanly shutdown the application and clean up any leftovers in the system in Application::quit()